packages/camera/camera_windows/windows/device_info.h

```diff
@@ -0,0 +1,25 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef PACKAGES_CAMERA_CAMERA_WINDOWS_WINDOWS_DEVICE_INFO_H_
+#define PACKAGES_CAMERA_CAMERA_WINDOWS_WINDOWS_DEVICE_INFO_H_
+
+#include <memory>
+#include <string>
+
+namespace camera_windows {
+
+struct CaptureDeviceInfo {
+  std::string display_name;
+  std::string device_id;
+};
+
+std::string GetUniqueDeviceName(std::unique_ptr<CaptureDeviceInfo> device_info);
```

Comment:
Since this is just getting a device name and not transferring ownership or consuming the unique_ptr, consider just passing a const ref to CaptureDeviceInfo.

Alternatively, I wonder if it might make sense to make CaptureDeviceInfo a class and make this into a getter on it?

Reply:
Converted to a class